brave/app/scripts/metamask-controller.js

```diff
@@ -1,8 +1,35 @@
 const MetamaskController = require('../../../app/scripts/metamask-controller')
+const ProviderApprovalController = require('../../../app/scripts/controllers/provider-approval')
 const nodeify = require('../../../app/scripts/lib/nodeify')
 
 module.exports = class BraveController extends MetamaskController {
 
+  constructor (opts) {
+    super(opts)
+
+    this.optsOpenPopup = opts.openPopup
+    this.providerApprovalController = new ProviderApprovalController({
+      closePopup: opts.closePopup,
+      openPopup: this.openPopup.bind(this),
+      keyringController: this.keyringController,
+      preferencesController: this.preferencesController,
+    })
+
+    this.memStore.config.ProviderApprovalController = this.providerApprovalController.store
+    this.memStore.subscribe(this.sendUpdate.bind(this))
+  }
+
+  openPopup () {
+    const prefStore = this.preferencesController.store
+    const hasOnboarded = prefStore.getState().completedOnboarding
+
+    if (hasOnboarded) {
+      this.optsOpenPopup()
+    } else {
+      chrome.tabs.create({ url: 'home.html' })
```

Comment:
I think this URL should be chrome://wallet/home.html

Reply:
@bbondy won't the URL rewrite core-side catch and redirect it?

Reply:
MM creates windows/tabs using relative paths

Reply:
You can try, I'm ok if it works. I had seen that if you try to load the chrome-extension URL directly it doesn't show the Brave icon and verified brave page, but if you do use the chrome:// URL which shows as brave:// it does.  If it works like you have it oh master though, then that's cool.

Reply:
Updated this to `chrome://wallet/home.html`, seems that you don't get the brave page items without it, as you said.

One thing to keep in mind is that MM does use the relative path to open windows and views, so this may be something to go through and update across the board as well.